text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+4. `KEYS_ONLY`: Only the key attributes of the modified item are written to the stream.
+
+This would be the same for each replica. By default, streams are not enabled for global table. But, if
+you have a replica present which is not of the same region as where the stack is deployed to, then the
+stream type would be set to `NEW_AND_OLD_IMAGES` automatically since it is required for table replication.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    stream: StreamViewType.NEW_IMAGE,
+  });
+```
+
+#### Time To Live
+
+You can enable time to live for an attribute of the global table. This would be the same for each replica.
+By default, it is not enabled for any attributes.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    timeToLiveAttribute: 'Foo',
+  });
+```
+
+#### Replicas
+
+You can define replicas for your global table. By default, if this configuration is undefined, then the
+table is deployed to the region where stack is being deployed.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-east-1',
+      },
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+```
+
+There are per replica configurations that are available. If any of these are undefined in the replica and
+defined in the table level props, then those are copied over to the replica configuration.
+But, if the replica has any configuration that is defined and the tabel level properties are defined as well,
+then the replica property value takes precedence and would be assigned.
+
+Options:
+
+* `region` --> Needs to be specified
+* `contributorInsightsEnabled` --> Gets copied over from table level props if defined.
+* `deletionProtection` --> Gets copied over from table level props if defined.
+* `pointInTimeRecovery` --> Gets copied over from table level props if defined.
+* `tableClass` --> Gets copied over from table level props if defined.
+* `tags` --> Gets copied over from table level props if defined.
+* `readWithAutoScaling` --> Gets copied over from table level props if defined.
+* `readCapactity` --> Gets copied over from table level props if defined.
+* `encryptionKey` --> This needs to be defined per replica
+* `kinesisStream` --> This needs to be defined per replica
+* `gsiProps`
+  * `indexName` --> Needs to be specified
+  * `contributorInsightsEnabled` --> Gets copied over from table level props or replica level props if defined.
+
+You can also add a replica using `addReplica` method.
+
+```typescript
+  globalTable.addReplica({
+    region: 'us-east-1',
+  });
+```
+
+__NOTE:__
+
+* A replica needs to be deployed to the region where stack is deployed to. We do this automatically if
+user has not defined the region in replica.
```

Comment:
I would say we always add the current region automatically, and the user only specifies additional replication regions in their input.